.agent/src/jinx/runner.py

```
@@ -801,3 +910,44 @@ def _slice_file_content(result_content: str, params: Dict[str, Any]) -> str:
+def _read_stdin_line(timeout: int) -> Optional[str]:
+    """Read a line from stdin with a timeout (seconds).
+
+    Returns the line (without trailing newline) or None if timeout elapsed.
+    """
+    q: _queue.Queue = _queue.Queue()
+
+    def _reader(queue: _queue.Queue):
+        try:
+            line = sys.stdin.readline()
+            queue.put(line)
+        except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)